app/src/main/java/de/danoeh/antennapod/ui/screen/playback/TranscriptFragment.java

```diff
@@ -0,0 +1,290 @@
+                ((FeedMedia) this.media).setTranscript(transcript);
+
+                if (transcript != null) {
+                    segmentsMap = transcript.getSegmentsMap();
+                }
+                emitter.onSuccess(media);
+            } else {
+                emitter.onComplete();
+            }
+        })
+        .subscribeOn(Schedulers.io())
+        .observeOn(AndroidSchedulers.mainThread())
+        .subscribe(media -> onMediaChanged((Playable) media),
+                error -> Log.e(TAG, Log.getStackTraceString(error)));
+    }
+
+    private void onMediaChanged(Playable media) {
+        if (media == null || ! (media instanceof FeedMedia)) {
+            return;
+        }
+        this.media = media;
+
+        FeedMedia feedMedia = (FeedMedia) media;
+        if (!feedMedia.hasTranscript()) {
+            dismiss();
+            Toast.makeText(getContext(), R.string.no_transcript_label, Toast.LENGTH_LONG).show();
+            return;
+        }
+
+        progressBar.setVisibility(View.GONE);
+        adapter.setMedia(media);
+        ((AlertDialog) getDialog()).getButton(DialogInterface.BUTTON_NEUTRAL).setVisibility(View.INVISIBLE);
+        if (!TextUtils.isEmpty(((FeedMedia) media).getItem().getPodcastIndexTranscriptUrl())) {
+            ((AlertDialog) getDialog()).getButton(DialogInterface.BUTTON_NEUTRAL).setVisibility(View.VISIBLE);
+        }
+    }
+
+    @Override
+    public void onPause() {
+        super.onPause();
+    }
+
+    public void scrollToPlayPosition(long playPosition) {
+        if (segmentsMap == null) {
+            return;
+        }
+        Map.Entry<Long, TranscriptSegment> entry = segmentsMap.floorEntry(playPosition);
+        if (entry != null) {
+            if (transcript == null) {
+                return;
+            }
+            Integer pos = transcript.getIndex(entry);
+            Log.d(TAG, "scrollToPlayPosition " + playPosition + " RV pos" + pos);
+            scrollToPosition(pos);
+        }
+    }
+
+    public void scrollToPosition(Integer pos) {
+        RecyclerView rv = viewBinding.transcriptList;
+        if (rv == null) {
+            return;
+        }
+        if (pos == null) {
+            return;
+        }
+
+        final LinearSmoothScroller smoothScroller = new LinearSmoothScroller(getActivity()) {
+            @Override
+            protected int getVerticalSnapPreference() {
+                return LinearSmoothScroller.SNAP_TO_START;
+            }
+        };
+        if (pos > 0) {
+            smoothScroller.setTargetPosition(pos - 1);  // pos on which item you want to scroll recycler view
+            rv.getLayoutManager().startSmoothScroll(smoothScroller);
+        }
+
+        RecyclerView.LayoutManager lm = rv.getLayoutManager();
+        if (lm == null) {
+            return;
+        }
+        View v = lm.findViewByPosition(pos);
+        if (v == null) {
+            return;
+        }
```

Comment:
This feels a bit too careful :) So many checks make the code harder to read

Reply:
removed most of the useless checks